server/src/main/java/org/apache/cassandra/sidecar/db/schema/TableHistorySchema.java

```diff
@@ -0,0 +1,168 @@
+ *       chronological access to schema changes</li>
+ *   <li><strong>Versioning:</strong> Uses UUID-based versioning for unique schema identification</li>
+ *   <li><strong>Timestamping:</strong> Automatic creation timestamp tracking with {@code created_at}</li>
+ * </ul>
+ * <p>
+ * The table structure includes:
+ * <pre>{@code
+ * CREATE TABLE table_schema_history (
+ *   ks text,                    -- Keyspace name
+ *   tb text,                    -- Table name
+ *   version uuid,               -- Schema version identifier
+ *   created_at timeuuid,        -- Schema creation timestamp
+ *   table_schema text,          -- Complete table schema DDL
+ *   PRIMARY KEY ((ks, tb), version)
+ * )
+ * }</pre>
+ * <p>
+ * This schema supports CDC operations by maintaining a complete history of table schemas,
+ * allowing CDC consumers to process data with the correct schema version that was active
+ * when the data was written. This is crucial for maintaining data integrity across schema
+ * evolution in long-running CDC pipelines.
+ * <p>
+ * This class is thread-safe and designed as a singleton for dependency injection into
+ * components that require table schema history access.
+ *
+ * @see TableSchema
+ * @see org.apache.cassandra.sidecar.db.CdcDatabaseAccessor
+ */
+@Singleton
+public class TableHistorySchema extends TableSchema
+{
+    private static final String TABLE_SCHEMA_HISTORY = "table_schema_history";
+
+    private final SchemaKeyspaceConfiguration keyspaceConfig;
+
+    // prepared statements
+    private PreparedStatement insertTableSchema;
+    private PreparedStatement selectVersionTableSchema;
+
+    @Inject
+    public TableHistorySchema(ServiceConfiguration configuration)
+    {
+        this.keyspaceConfig = configuration.schemaKeyspaceConfiguration();
+    }
+
+    @Override
+    protected void prepareStatements(@NotNull Session session)
+    {
+        insertTableSchema = prepare(insertTableSchema, session, CqlLiterals.insertTableSchema(keyspaceConfig));
+        selectVersionTableSchema = prepare(selectVersionTableSchema, session, CqlLiterals.selectVersionTableSchema(keyspaceConfig));
+    }
+
+    @Override
+    protected String keyspaceName()
+    {
+        return keyspaceConfig.keyspace();
+    }
+
+    @Override
+    protected String tableName()
+    {
+        return TABLE_SCHEMA_HISTORY;
+    }
+
+    @Override
+    protected boolean exists(@NotNull Metadata metadata)
+    {
+        KeyspaceMetadata ksMetadata = metadata.getKeyspace(keyspaceConfig.keyspace());
+        if (ksMetadata == null)
+        {
+            return false;
+        }
+
+        return ksMetadata.getTable(TABLE_SCHEMA_HISTORY) != null;
+    }
+
+    @Override
+    protected String createSchemaStatement()
+    {
+        return String.format("CREATE TABLE IF NOT EXISTS %s.%s (" +
+                             "  ks text," +
+                             "  tb text," +
```

Comment:
NIT: to maintain consistency with Cassandra column naming
```suggestion
                             "  keyspace_name text," +
                             "  table_name text," +
```